streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -237,7 +237,7 @@ private[stream] trait ZStreamPlatformSpecificConstructors {
             ZStream.repeatZIOChunkOption(
               for {
                 bytesRead <- ZIO.attempt(channel.read(reusableBuffer)).asSomeError
-                _         <- ZIO.fail(None).when(bytesRead == -1)
+                _         <- Exit.failNone.when(bytesRead == -1)
```

Comment:
A bit outside of the scope of this PR, but since we're at it might change these usages to `whenDiscard`

Reply:
Done